### PR TITLE
WIP: change API of envelope_correlation

### PR DESCRIPTION
**Pull request by agramfort**
- add callable option to combine
- make mean the default and not median

still todo is be able to disable the orthogonalization

### CODE REVIEW

File: mne/connectivity/envelope.py [hunk 1]
```diff
@@ -51,7 +55,8 @@ def envelope_correlation(data, combine='median', verbose=None):
     from scipy.signal import hilbert
     corrs = list()
     n_nodes = None
-    _validate_type(combine, (str, None), 'combine')
+    if not callable(combine):
+        _validate_type(combine, (str, None), 'combine')
```
Comment (larsoner) [inline mne/connectivity/envelope.py]: This should be refactored with the code in `evoked` (or wherever we recently added `trim_mean` support, etc.) with a private function / the same set of options.

**Comment by agramfort** [comment]
done

>

File: mne/epochs.py [hunk 2]
```diff
@@ -834,21 +835,7 @@ def _compute_aggregate(self, picks, mode='mean'):
...
-                    return np.median(data, axis=0)
-            elif mode == "std":
-                def fun(data):
-                    return np.std(data, axis=0)
-            elif callable(mode):
-                fun = mode
-            else:
-                raise ValueError("mode must be mean, median, std, or callable"
-                                 ", got %s (type %s)." % (mode, type(mode)))
+            fun = _check_combine(mode, valid=('mean', 'median', 'std'))
```
Comment (larsoner) [inline mne/epochs.py]: no need to provide these `kwargs` because they are the default, but it's okay to leave I guess

**Comment by larsoner** [comment]
MRG or still WIP? Happy to merge now and make non-orth a separate PR

**Pull request merged**